docs: Add an existing Server (#548)
devopness/devopness: docs/servers/add-existing-server.md

---
title: Add an existing Server

> intro: If you have servers provisioned manually or by other processes you can still connect them to your Devopness account and take the most of the Devopness infrastructure management features.

Reviewer: Don't know the best way to put it on but i believe its important to have in the documentation that we don't support / recommend to add servers already used for any real operations on the user environment.

Author: @plantaz it seems to me that will belong to deeper level overview guides to be written in the future, whereas the current docs are simple steps to achieve a certain task, not being the steps responsible for educating the users on best practices. Does it make sense to you?

Reviewer: Maybe we can change "If you have servers provisioned manually or by other processes" to "If you have servers provisioned outside of your desired environment at Devopness" because our users can add servers provisioned by Devopness on others environments as a self-hosted.

Reviewer: Yes, makes sense! I just commented because in this case in particular we cannot guarantee that the server setup will not "break" any of their operation on this server. But maybe, as you suggested, will fit better in another place.

links:
    overview:
    quickstart:
    previous:
    next:
    guides:
    related:
    featured:
---

# Add an existing Server
1. On Devopness, navigate to a project then select an environment
2. If there's no other server in the selected environment, just click `Add Server`
3. If a list of resource cards is displayed, find the `Servers` card
4. Click `View` in the `Servers` card to see a list of existing `Servers`
5. On the upper-right corner of the list click `ADD SERVER`

> 6. On the `Cloud Provider` drop-down menu choose `VPS/Self Hosted`

Reviewer: ```suggestion
6. On the `Cloud Provider` drop-down menu choose `VPS/Self Hosted`
    - Some configuration needs to be done when a server is added and we currently do not support backing up the previous configuration that will be overwritten
```

7. Provide the `Public IP address` of your existing server

> 6. Enter the `SSH Port` to be used to connect to the server

Author: @plantaz it seems more logical to warn user about the risk of adding an existing server after the `IP address` has been informed. Do you agree?
```suggestion
6. Enter the `SSH Port` to be used to connect to the server
> Note: we do not recommend connecting servers that are already being used in production. Server's existing configuration will be lost and replaced by Devopness during the server setup process
```

7. Click `NEXT`
8. Follow the prompts to complete the configuration of the new `Server`
9. Click `CONFIRM`
    - If you see a message saying `Error establishing a SSH connection to server`, please verify if the IP address and SSH port used are correct and if external SSH connections are allowed on this server
10. After the `Server` is added you will be redirected to the `Connect` page
11. Find the `Command to connect` field and click `Copy to clipboard`
12. Log into your server as the `root` user
13. Paste the command that has been copied to the clipboard and run it

> 14. A message will be displayed in the terminal window when the server is connected to Devopness

Reviewer: ```suggestion
14. A message will be displayed in the terminal window when the server connects to Devopness
```

Author: @plantaz as the server itself does not connect, what about?
```suggestion
14. A message will be displayed in the terminal window when the server connect command is successfully completed
```

> 15. Get back to Devopness and start managing your server! :)

Reviewer: ```suggestion
15. On Devopness, the servers `Connect` page will confirm the connection
```

Author: @plantaz the page does not confirm the connection, what about?
```suggestion
15. On Devopness, once the server is displayed as `Connected`, it's been added to the selected environment
```
